Plugins/PluginWebParser/WebParser.cpp

```diff
@@ -1511,28 +1512,9 @@ BYTE* DownloadUrl(HINTERNET handle, std::wstring& url, DWORD* dataSize, bool for
 			return nullptr;
 		}
 
-		FILE* file = _wfopen(path, L"rb");
-		if (!file)
-		{
-			return nullptr;
-		}
-
-		fseek(file, 0, SEEK_END);
-		*dataSize = ftell(file);
-		rewind(file);
-
-		BYTE* buffer = (BYTE*)malloc(*dataSize + 3);
-		if (buffer)
-		{
-			fread(buffer, 1, *dataSize, file);
-
-			// Triple null terminate the buffer.
-			buffer[*dataSize] = 0;
-			buffer[*dataSize + 1] = 0;
-			buffer[*dataSize + 2] = 0;
-		}
+		BYTE* buffer = FileUtil::ReadFullFile(path).release();
+		*dataSize = strlen((char*)buffer);

```

Comment:
This is problematic if the file contains embedded NULL characters (because `strlen` stops at the first NULL). Could you instead add an optional `size` parameter to `ReadFullFile`? That is, something like `std::unique_ptr<BYTE[]> ReadFullFile(const std::wstring& path, size_t* size = nullptr)`.